homeassistant/helpers/template.py

```diff
@@ -217,24 +258,41 @@ def __init__(self, hass):
 
     def __getattr__(self, name):
         """Return the domain state."""
+        if '.' in name:
```

Comment:
this is only possible when people use `getattr(obj, 'light.kitchen')`, which i don't think is possible inside template?

Reply:
```states['light.kitchen'].state```  will follow this code path.

Reply:
```states('light.kitchen')``` is different in that it gets the state value rather than the object. If you want to say access attributes, you have to use the [] version. Without this change the Domain object gets returned, which is different in that it's a collection.